cypress/e2e/gui-tests/gui-customers.cy.js

```diff
@@ -0,0 +1,60 @@
+describe('Customers', () => {
+    beforeEach(() => {
+      cy.visit('/')
+    });
+  
+    context('Filtering Customers', () => {
+      it('filters and shows only Small customers', () => { 
+        cy.get('[data-testid="filter"]').select('Small');
+        cy.wait(1500); // eslint-disable-line
+        cy.get('[data-testid="table"] tbody tr').each(($row) => {
+          cy.wrap($row).find('td').eq(3).should('contain.text', 'Small');
+        });
+      });
+  
+      it('filters and shows only Medium customers', () => { 
+        cy.get('[data-testid="filter"]').select('Medium');
+        cy.wait(1500); // eslint-disable-line
+        cy.get('[data-testid="table"] tbody tr').each(($row) => {
+          cy.wrap($row).find('td').eq(3).should('contain.text', 'Medium');
+        });
+      });
+  
+      it('filters and shows only Enterprise customers', () => { 
+        cy.get('[data-testid="filter"]').select('Enterprise');
+        cy.wait(1500); // eslint-disable-line
+        cy.get('[data-testid="table"] tbody tr').each(($row) => {
+          cy.wrap($row).find('td').eq(3).should('contain.text', 'Enterprise');
+        });
+      });
+  
+      it('filters and shows only Large Enterprise customers', () => { 
+        cy.get('[data-testid="filter"]').select('Large Enterprise');
+        cy.wait(1500); // eslint-disable-line
+        cy.get('[data-testid="table"] tbody tr').each(($row) => {
+          cy.wrap($row).find('td').eq(3).should('contain.text', 'Large Enterprise');
+        });
+      });
+  
+      it('filters and shows only Very Large Enterprise customers', () => { 
+        cy.get('[data-testid="filter"]').select('Very Large Enterprise');
+        cy.wait(1500); // eslint-disable-line
+        cy.get('[data-testid="table"] tbody tr').each(($row) => {
+          cy.wrap($row).find('td').eq(3).should('contain.text', 'Very Large Enterprise');
+        });
+      });
+    });
+  
+    context('No Customers Available', () => {
+      it('should display "No customers available." message when there are no customers in the database', () => {
+        cy.simulateNoCustomers();
+        cy.get('span').should('have.text', 'No customers available.');
```

Comment:
Que tal algo assim?

```suggestion
        cy.contains('span', 'No customers available.').should('be.visible')
```

Pois sua verificação só garante a existência do elemento no DOM, não sua visibilidade.

Um elemento pode existir e não estar visível e o teste passar.

Já se testo sua visibibilidade (`.should('be.visible')`) e um elemento existe no DOM, mas não está visível, o teste ira falhar.